models/src/main/scala/coop/rchain/models/rholang/sorter/ReceiveSortMatcher.scala

```diff
@@ -29,15 +29,17 @@ object ReceiveSortMatcher extends Sortable[Receive] {
   // This function will then sort the insides of the preordered binds.
   def sortMatch[F[_]: Sync](r: Receive): F[ScoredTerm[Receive]] =
     for {
-      sortedBinds         <- r.binds.toList.traverse(sortBind[F])
-      persistentScore     = if (r.persistent) 1L else 0L
+      sortedBinds     <- r.binds.toList.traverse(sortBind[F])
+      persistentScore = if (r.persistent) 1L else 0L
+      // TODO: score peek
```

Comment:
is there a ticket for this?

Reply:
There is now https://rchain.atlassian.net/browse/RCHAIN-3600

Reply:
also, I think there are tests for the sorter / normalizer that check that fields are not omitted from the input. We need a similar test for peek in the reduce

Reply:
Not sure which test you have in mind here

Reply:
The one in SortTest:251. You should set `peek = true` in s2.

Also, make sure the property specs in that class (search: forAll) fail without your changes to the sorter. They should start failing as soon as you added the `peek` field. If they don't, we need to tweak the implicit PropertyCheckConfiguration (see scalatest docs).

Reply:
I think you meant `SortTest:387`, as peek is defined on the `Receive` level.

As for the generators, this can be done in a separate PR